src/aks-preview/azext_aks_preview/_helpers.py

```diff
@@ -49,9 +50,29 @@ def _set_vm_set_type(vm_set_type, kubernetes_version):
 
 
 def _set_outbound_type(outbound_type, vnet_subnet_id, load_balancer_sku, load_balancer_profile):
-    if outbound_type != CONST_OUTBOUND_TYPE_USER_DEFINED_ROUTING:
+    if (
+        outbound_type != CONST_OUTBOUND_TYPE_USER_DEFINED_ROUTING and
+        outbound_type != CONST_OUTBOUND_TYPE_MANAGED_NAT_GATEWAY and
+        outbound_type != CONST_OUTBOUND_TYPE_USER_ASSIGNED_NAT_GATEWAY
+    ):
         return CONST_OUTBOUND_TYPE_LOAD_BALANCER
 
+    if outbound_type == CONST_OUTBOUND_TYPE_MANAGED_NAT_GATEWAY:
+        if load_balancer_sku == "basic":
+            raise CLIError("managedNATGateway doesn't support basic load balancer sku")
+
+        return CONST_OUTBOUND_TYPE_MANAGED_NAT_GATEWAY
+
+    if outbound_type == CONST_OUTBOUND_TYPE_USER_ASSIGNED_NAT_GATEWAY:
+        if load_balancer_sku == "basic":
+            raise CLIError("userAssignedNATGateway doesn't support basic load balancer sku")
+
+        if vnet_subnet_id in ["", None]:
+            raise CLIError("--vnet-subnet-id must be specified for userAssignedNATGateway and it must \
+            be pre-associated with a NAT gateway with outbound public IPs or IP prefixes")
```

Comment:
Could you please use the specific error type ` ArgumentUsageError` instead of `CLIError` ?

Reply:
Updated.